docs: fix readme links after readme structure changed to be in a docs folder

MichaelDimmitt opened the pull request:
# Summary

Fixed all of the broken links in the readme that occured after the folder structure for the project changed.

Fixes: #1155 
See also: #456 , #492 

## Other Information

I noticed that docsify was used for the translations? 

Are the english docs and any other docs auto generated?

If that is the case that change may need to be added in the auto generated docs.

#### Happy to provide any additional info

Stratus3D commented:
Thanks for the PR @MichaelDimmitt ! Sorry the late response. Your PR is next on my list and I'll hopefully review it tomorrow. Also, not important yet, but some recent changes on the master branch have introduced a conflict here.

Stratus3D commented on README.md:
```diff
@@ -27,7 +27,7 @@ asdf is a CLI tool that can manage multiple language runtime versions on a per-p
 ## Contributing
 
-See [CONTRIBUTING.md in the repo](https://github.com/asdf-vm/asdf/blob/master/CONTRIBUTING.md) or the [Contributing section on the docs site](http://asdf-vm.github.io/asdf/#/contributing-core-asdf).
+See [CONTRIBUTING.md in the repo](https://github.com/asdf-vm/asdf/blob/master/CONTRIBUTING.md) or the [Contributing section on the docs site](https://asdf-vm.github.io/asdf/#/contributing-core-asdf).
```
Thanks! Looks good.

Stratus3D commented on docs/guide/getting-started.md:
````diff
@@ -306,7 +306,7 @@ asdf global nodejs latest
 ```
 
-Some OSs already have tools installed that are managed by the system and not `asdf`, `python` is a common example. You need to tell `asdf` to pass the management back to the system. The [Versions reference section](/manage/versions.md) will guide you.
+Some OSs already have tools installed that are managed by the system and not `asdf`, `python` is a common example. You need to tell `asdf` to pass the management back to the system. The [Versions reference section](/docs/manage/versions.md) will guide you.
````
It's not clear to my why this `/docs/` prefix is needed. I believe the canonical URL is `https://asdf-vm.com/manage/versions.html` so I would think that would translate to `/manage/versions.md` here.

MichaelDimmitt replied:
When I navigate here: 
https://github.com/asdf-vm/asdf/tree/master/docs/manage/versions.md
success

When I navigate here:
https://github.com/asdf-vm/asdf/tree/master/manage/versions.md
404 not found

MichaelDimmitt replied:
I may need a quick

How to on what tool is used if the docs are generated.
as these are hardcoded changes to the Readme files on github.

@Stratus3D , thanks for taking a look at this!

Stratus3D replied:
Please read this page - http://asdf-vm.com/contribute/documentation.html#initial-setup

The files in the `docs/` directory are used by vuepress to generate the documentation website. The links will not work when viewing the source code on Github.

MichaelDimmitt replied:
So the pr should just be closed?

interacting with the docs on github is a helpful user experience since you can search across the entire repo

I will look at initial setup and see if there is anything I can do to fix github without breaking vuepress.

it might make sense for me to close the mr and make a new mr with just the https.

Stratus3D replied:
The only change I see here that I think we need is the `http` => `https` in the readme. All other changes can be discarded.

Stratus3D commented on docs/pt-br/manage/versions.md:
```diff
@@ -86,7 +86,7 @@ asdf local <name> latest[:<version>]
 `local` escreve a versão para `$PWD/.tool-versions`, crie se necessário .
 
-Veja em `.tool-versions` [arquivo de seleção de configuração](/pt-br/core-configuration) para mais detalhes.
+Veja em `.tool-versions` [arquivo de seleção de configuração](/docs/pt-br/core-configuration) para mais detalhes.
```
For all additions of the `/docs/` segment, same question as above. Even for other languages the `/docs/` segment isn't part of the canonical URL.

MichaelDimmitt commented:
Hi @Stratus3D , 

I figured out and added to the pr a change to the vuepress configuration

adding `base: "/docs/"`

allows the links to work for both github and the main website.

The vuepress deploy talks describe how it
https://vuepress.vuejs.org/guide/deploy.html#github-pages

a link where there was a very similar solution:
https://cexamples.com/examples/vue-js-how-to-use-route-from-index-html-to-docs-folder

MichaelDimmitt commented:
One thing is after building it may be valueable for a redirect in the index.html file.
http://localhost:8080/
correctly redirects to http://localhost:8080/docs

http://localhost:8080/docs/nope
correctly renders 404 page

routes with /docs/ 
work as expected for `en`, `zh-hans`, and `pt-br`
http://localhost:8080/docs/manage/core.html#installation-setup
http://localhost:8080/docs/zh-hans/manage/core.html
http://localhost:8080/docs/pt-br/manage/core.html#installation-setup

![Screen Shot 2022-02-05 at 3 40 53 PM](https://user-images.githubusercontent.com/11463275/152658224-5573fa66-a642-4798-9eec-3165b5ab46ca.png)

MichaelDimmitt commented:
## Final Note
Vuepress expects the files to be in a `/docs` directory since the base specifies `base: "/docs/"`

Therefore I added a file called SERVER_REDIRECT
The package.json copies this into .vuepress/dist as index.html after the build has completed.

This redirects the root directory `/` to the `/docs/` scope.

a copy of the 404 page is also moved from `/dist/docs/` to `/dist/` scope, 404 works, 
However it also changes from `/theroute-entered` to `/docs/theroute-entered` 
^ this was not intentional just noting it here. 

When you run vuepress locally it already does a redirect as it was expecting this in the developer workflow for github pages.

However, they do not have that functionality for the build command because they expected it to be embedded into an already working site like a branch off of a "Github Pages" 
https://vuepress.vuejs.org/guide/deploy.html#github-pages

Here is where I got the redirect html: https://www.w3docs.com/snippets/html/how-to-redirect-a-web-page-in-html.html#:~:text=The%20simplest%20way%20to%20redirect,seconds%20for%20the%20content%20attribute.

MichaelDimmitt commented:
## Sorry that turned out to be kind of complex but it works for vuepress, 

## Check it out. 🎉
I see there are other locale pr's in flight
I can update this branch if those pr's are merged first.

Regards, Michael Dimmitt

MichaelDimmitt commented:
@Stratus3D , I put just the http to https change in a new pr
https://github.com/asdf-vm/asdf/pull/1169

so that you do not need to merge this one if you do not like it.

Stratus3D commented:
@jthegedus can you take a look at this PR? I'm not very familiar with vuepress.

jthegedus commented:
I will take a look at this, but the redirect to `http` could probably just be fixed by enabling "enforce HTTPS" in our GH Pages settings:

![image](https://user-images.githubusercontent.com/20798510/155081911-cfbcb350-6019-495a-8302-cf9d1e015ca1.png)

@MichaelDimmitt You say in the PR summary

>after the folder structure for the project changed.

Can you link the recent PR where the folder structure changed?

MichaelDimmitt commented:
This is the pr that caused links to stop working on github
https://github.com/asdf-vm/asdf/pull/1000 

Thanks for your help on this @jthegedus

MichaelDimmitt commented:
@jthegedus , any way I can help test this ?

Feel free to push changes to the pr
i.e. removing the redirect file.

MichaelDimmitt commented:
@jthegedus is there anything blocking this pr?
happy to help

Stratus3D commented:
@MichaelDimmitt I'm letting @jthegedus handle this one. I'm not the expert on vuepress and docs so I don't want to make the decision here.

Stratus3D commented:
Closing for now as I believe the changes we need here are in another PR by @MichaelDimmitt that has already been merged.
